runtime/sys.c

```diff
@@ -319,6 +319,36 @@ CAMLprim value caml_sys_chdir(value dirname)
   CAMLreturn(Val_unit);
 }
 
+CAMLprim value caml_sys_mkdir(value path, value perm)
+{
+  CAMLparam2(path, perm);
+  char_os * p;
+  int ret;
+  caml_sys_check_path(path);
+  p = caml_stat_strdup_to_os(String_val(path));
+  caml_enter_blocking_section();
+  ret = mkdir_os(p, Int_val(perm));
+  caml_leave_blocking_section();
+  caml_stat_free(p);
+  if (ret == -1) caml_sys_error(path);
+  CAMLreturn(Val_unit);
+}
```

Comment:
As before, this is very similar to `otherlibs/unix/mkdir.c` (which has been merged with `otherlibs/win32unix/mkdir.c` as part of this PR):
```diff
-CAMLprim value caml_sys_mkdir(value path, value perm)
+CAMLprim value unix_mkdir(value path, value perm)
 {
   CAMLparam2(path, perm);
   char_os * p;
   int ret;
-  caml_sys_check_path(path);
+  caml_unix_check_path(path, "mkdir");
   p = caml_stat_strdup_to_os(String_val(path));
   caml_enter_blocking_section();
   ret = mkdir_os(p, Int_val(perm));
   caml_leave_blocking_section();
   caml_stat_free(p);
-  if (ret == -1) caml_sys_error(path);
+  if (ret == -1) uerror("mkdir", path);
   CAMLreturn(Val_unit);
 }
```